### DISCUSSION TITLE
Blurred album cover as lyrics background

### DISCUSSION THREAD

**Discussion by SunsetTechuila** [nimsandu/spicetify-bloom#205] [Show and tell]
What do you think? I'm not sure if it is better than it was
![Без имени-5](https://user-images.githubusercontent.com/115353812/221914151-f5bd9c8b-b1b5-456d-86b5-e6f0084d0d9c.png)
Just PoC, noting more

**Comment by SunsetTechuila** [comment]
![Без имени-6](https://user-images.githubusercontent.com/115353812/221945882-e9b1ed4c-3806-4fc8-b58e-6c4c8089251c.png)

**Comment by SunsetTechuila** [comment]
https://github.com/SunsetTechuila/spicetify-bloom/tree/lyrics-test - test version
Will be waiting for response

**Comment by nimsandu** [comment]
@SunsetTechuila Well that looks good.
But my personal belief is that lyrics page is better when it's readability is higher and more cleaner. If someone needs a better lyrics page then it can be fulfilled by lyrics extension and full screen extension. 
So what I believe is that we should keep the original lyrics page cleaner for users who are focused in readability rather than stunning ui.

Please note that this is my personal opinion and all bloomers' beliefs are welcome. Only those suggestions will be considered rather than mine

**Reply by SunsetTechuila** [reply]
Sometimes it is even more readable than the original:
![image](https://user-images.githubusercontent.com/115353812/222112527-7c15770b-a78b-4219-8739-e91feae5e5e2.png)
![image](https://user-images.githubusercontent.com/115353812/222112584-c8024d01-e83e-4d64-bb74-b9c35839945e.png)

Try the test version please when you have time

**Reply by nimsandu** [reply]
@SunsetTechuila Sure, ill test it out asap.
Thanks

**Comment by nimsandu** [comment]
@SunsetTechuila i have tested your implementation and i immediately started to love it.
i have few suggestions regarding your work. 
can we change 

- background blend mode form` darken` to `color-burn`  this improve _the readability with light color album arts_
- change saturation from `0.8` to `2` _this makes background more pleasant while blend mode is set to `color-burn`_ 

![image](https://user-images.githubusercontent.com/57834844/222152477-27abce12-27e5-429e-8c29-e17346353dc6.png)

### **preview 1**
### this is the original view 
![image](https://user-images.githubusercontent.com/57834844/222152261-69124665-2a1a-4509-bc1e-7a14820bdf47.png)
### this is the new look
![image](https://user-images.githubusercontent.com/57834844/222156777-c1dc5e32-62a6-4fe0-8bd1-c06670db9b45.png)



### **preview 2**
### this is the original view 
![image](https://user-images.githubusercontent.com/57834844/222153553-14347a28-a453-4da8-a6bc-af34f7e6551e.png)

### this is the new look
![image](https://user-images.githubusercontent.com/57834844/222153589-2ea90ff0-e16b-4be8-b834-23ed8d53b546.png)

### **preview 3**
### this is the original view 
![image](https://user-images.githubusercontent.com/57834844/222157177-3834d1b9-2ea8-469a-96fe-c60b21d32e25.png)

### this is the new look
![image](https://user-images.githubusercontent.com/57834844/222157203-7df95113-399e-4f23-9953-5819cf149b2c.png)

@SunsetTechuila what do you think?

**Reply by SunsetTechuila** [reply]
Glad to hear that you liked it, so I'll do the optimization and animations and let you know when I'm done.
About your suggestions - they are really cool, a bit less saturation and contrast and would be perfect, thanks

**Reply by nimsandu** [reply]
> Glad to hear that you liked it, so I'll do the optimization and animations and let you know when I'm done.
> 
> About your suggestions - they are really cool, a bit less saturation and contrast and will be perfect, thanks 

Roger that

**Reply by SunsetTechuila** [reply]
Ah, so color-burn itself causes excessive contrast, this is the algorithm. I can suggest to change background-color to #808080 instead . Apple Music has about the same brightness threshold (50% brightness on hsb)

**Reply by SunsetTechuila** [reply]
or we could just add ```brightness(0.7)``` to filters, I have no idea how to make bright colors darker keeping the saturation

**Reply by SunsetTechuila** [reply]
We definitely can't use color-burn because it makes artwork **too** dark sometimes, play [this album](https://open.spotify.com/album/0Ypo27qP4udOQKGfNT48hn?si=pDTJsS8rSfKNWdx4bALW_A). I tried to add ```contrast(0.9)```, but it still looks bad.
It is really hard to find universal solution

**Comment by SunsetTechuila** [comment]
and I question - what will be better in your opinion - keep the backdrop scrollable or make it fixed?

**Reply by SunsetTechuila** [reply]
made it fixed btw

**Comment by SunsetTechuila** [comment]
@nimsandu I've fixed lags, but I'm pretty sure I did everything stupid. If you know someone who can rewrite my code, please ask him. It works fine, but still

**Comment by nimsandu** [comment]
@SunsetTechuila I don't see anything stupid btw

**Reply by SunsetTechuila** [reply]
The backdrop is inserted before the os-padding element, so it is always visible, which is why I added visibility changing. It doesn't look like good design

**Reply by SunsetTechuila** [reply]
I rewrote the code using a separate MutationObserver for the right player bar panel, but the rest is unchanged

**Reply by SunsetTechuila** [reply]
If you have any ideas how to make backdrop sticky inside lyrics container - tell me please

**Comment by nimsandu** [comment]
> and I question - what will be better in your opinion - keep the backdrop scrollable or make it fixed?

Fixed is better ig

**Comment by SunsetTechuila** [comment]
@nimsandu branch has been updated - new animations, some fixes and improvements
![image](https://user-images.githubusercontent.com/115353812/222751442-4d5395a4-45e2-46d5-86b1-6bcfdc6e9903.png)

**Reply by nimsandu** [reply]
i have tested it out. everything works really nice 🙂. 
don't you think the now paying line animation can be more enhanced btw?
just a suggestion!
everything works and looks nice.
great work @SunsetTechuila

**Reply by SunsetTechuila** [reply]
What do you mean by more enhanced?

**Reply by nimsandu** [reply]
@SunsetTechuila Like smoothing the animation little bit.
I felt it kinda crispy.
But super cool ! Bring up the pr. We can do Improvements later. Nice work 😊. Wait for the pr

**Reply by SunsetTechuila** [reply]
PR will be within a week, there's still a lot of work to be done.

**Reply by SunsetTechuila** [reply]
Unless I jump out the window out of anger that nothing is working, of course)

**Reply by nimsandu** [reply]
Don't jump out from the window unless you can survive😂!
We need you.
Seriously don't panic or miss your personal works. Take your time. It all yours.

**Reply by SunsetTechuila** [reply]
> I felt it kinda crispy.

try now

**Reply by SunsetTechuila** [reply]
Yeah, it seems that the animations for lyrics have been completed. Whew 🫠
Now it's time for animations for the lyrics backdrop

**Reply by SunsetTechuila** [reply]
> Now it's time for animations for the lyrics backdrop

Test variant [added](https://github.com/SunsetTechuila/spicetify-bloom/commit/8e413a57bbd75ed0619c05df0c2f5016a239dd03)

**Comment by SunsetTechuila** [comment]
@nimsandu please test the current version

**Reply by nimsandu** [reply]
@SunsetTechuila im on it and sorry for keep you waiting

**Reply by nimsandu** [reply]
@SunsetTechuila i have trouble trying out your variant,
could you please update the branch with latest fixes and send me ?

**Reply by SunsetTechuila** [reply]
Synced with your main branch. Or did you mean something else?

**Reply by nimsandu** [reply]
yes i meant it! did you sync?

**Reply by SunsetTechuila** [reply]
Yes

**Reply by nimsandu** [reply]
![image](https://user-images.githubusercontent.com/57834844/223773603-dfd73eb3-7e47-42a4-afe2-c6708058d221.png)
no matter what i do i keep getting this other than your creation! still couldn't understand why!

**Reply by SunsetTechuila** [reply]
Did you edit user.css too?

**Reply by nimsandu** [reply]
yes i did.

**Reply by nimsandu** [reply]
ill start over and check again asap

**Reply by SunsetTechuila** [reply]
maybe wrong branch then?  You need https://github.com/SunsetTechuila/spicetify-bloom/tree/lyrics-test

**Comment by nimsandu** [comment]
@SunsetTechuila it works!
i mean its a w[o^(10^100)]w!
the transition is brilliant!
everything works fine!

**Comment by nimsandu** [comment]
@SunsetTechuila what you think about this configs 
```
lyrics-lyricsContent-lyric:active { transform: scale(0.97); transition: transform 50ms cubic-bezier(1, 0.04, 1, 1) 0s; }
.lyrics-lyricsContent-lyric:active::after { transform: scale(1); transition: transform 50ms cubic-bezier(0.22, -0.02, 0.34, 0.91) 0s; }
.lyrics-lyricsContent-highlight, .C3pBU1DsOUJJOAv89ZFT { color: rgba(255, 255, 255, 0.75) !important; }
.lyrics-lyricsContent-highlight.previous, .lyrics-lyricsContent-lyric.previous { animation: 450ms cubic-bezier(0.88, -0.01, 0.46, 1.04) 0s 1 normal forwards running lyric-previous; }
.lyrics-lyricsContent-unsynced, .lyrics-lyricsContent-active, .lyrics-lyricsContent-lyric:hover { color: rgb(255, 255, 255) !important; }
.lyrics-lyricsContent-active { animation: 400ms cubic-bezier(0.6, 0.19, 0.63, 0.95) 0s 1 normal forwards running lyric-active; }

```

**Reply by SunsetTechuila** [reply]
Will take a look

**Reply by SunsetTechuila** [reply]
the click animation is sharp and the acceleration of the lyric-previous and lyric-active animations at the end is unnecessary for my taste, sorry

**Reply by nimsandu** [reply]
@SunsetTechuila it's all right. cuz it's all yours!

**Reply by nimsandu** [reply]
love your work!

**Reply by SunsetTechuila** [reply]
so I can make PR?

**Reply by nimsandu** [reply]
yap.
everything looks good!
waiting for the pr.
it was a nice project with you. 
now it's time to for it to bloom.
thanks @SunsetTechuila !

**Comment by nimsandu** [comment]
#213 merged successfully!
enjoy new lyrics style while blooming!
credit @SunsetTechuila !
